AAT_gae/src/main/java/com/aat/restlet/AttendanceResource.java

```diff
@@ -72,8 +77,46 @@ public Representation create(Representation entity) {
 	}
 
 	@Put
-	public void update() {
-		// TODO: Implement
+	public String update(Representation entity) {
+		
+		// Check if of type Tutor
+		ResourceUtil.checkTokenPermissions(this, Tutor.class);
+		
+		// Get input parameters
+		Form params = new Form(entity);
+		Long studentId = Long.parseLong(getAttribute(Constants.userId), 10);
+		Long groupId = Long.parseLong(getAttribute(Constants.groupId), 10);
+		String flagMode = ResourceUtil.getParam(params, Constants.flagMode, true);
+		String dateWeek = ResourceUtil.getParam(params, Constants.dateWeek, true);
+		Date date;
+		AttendanceRecord attendance = retrieveAttendanceRecord(studentId,groupId);
+		List <Date> dates = null;
+		try {
+			date = new SimpleDateFormat("dd-MM-yyyy").parse(dateWeek);
+			
+			if (flagMode.equals("A")){
+					dates = attendance.getAttendance();
+			}
+			else if (flagMode.equals("P")){
+					dates = attendance.getPresentation();
+			}
+			else{
+				throw new ResourceException(409, "Conflict", "Invalid flag.", null);
+			}
+			
+			if (!dates.contains(date)){
```

Comment:
Shouldn't we also check to make sure the token matches correctly to the date in the date/token hashmap in the AttendanceRecord?